src/database-layer/db.ts

```diff
@@ -17,7 +17,10 @@ export interface CodedError extends Error {
 	code: number | string;
 }
 
-type CreateTransactionFn = (stackTraceErr?: Error) => Promise<Tx>;
+type CreateTransactionFn = (
+	stackTraceErr?: Error,
+	timeoutMS?: number,
+) => Promise<Tx>;
```

Comment:
We could also do this to avoid the parameter list on calls being unnecessarily long:
```typescript
options: {
    stackTraceErr?: Error,
    timeoutMS?: number,
}
```